utils/utils.go

```diff
@@ -0,0 +1,49 @@
+package utils
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+// BackupRequestBody returns a backup of the Request.Body, leaving it untouched
+// so the passed Request.Body will be still readable.
+func BackupRequestBody(req *http.Request) io.ReadCloser {
+	content, _ := ioutil.ReadAll(req.Body)
```

Comment:
Also can't you just use `ReadRequestAndRestore`? It seems overlapping.

Reply:
from docs:
```go
// BackupRequestBody returns a copy of the Request.Body.
func BackupRequestBody(*http.Request) (io.ReadCloser, error) {}

// ReadRequestAndRestore returns the content of the Response.Body, and restores it
func ReadRequestAndRestore(*http.Request) ([]byte, error) {}

```
so `BackupRequestBody` is used to get a backup that will be restored later, e.g in a `defer`,
and `ReadRequestAndRestore` is used to get the content of the bofy without ruining it.

Reply:
But since you could do as [here](https://github.com/src-d/metadata-retrieval/pull/57#discussion_r335419273), maybe you don't need `BackupRequestBody`?

Reply:
done following the alternative you suggested 2717914